### ISSUE TITLE
Running tests changes global git user name/email

### ISSUE THREAD

**Issue by jasonkarns** [sstephenson/rbenv#741]
After running the tests (`bats test/`), my global git user name and email is wrong. `test/--version.bats` sets the name and email to Tester <tester@test.local>.

https://github.com/sstephenson/rbenv/blob/5b9e4f05846f6bd03b09b8572142f53fd7a46e62/test/--version.bats#L7-L8

Am I running the tests incorrectly or is there some configuration that I should be doing to prevent this from happening? (Temporarily, I've just removed those lines from the file and have marked them as `--assume-unchanged` in the git index)

**Comment by mislav** [comment]
I'm sorry that happened. The [HOME value is supposed to be changed to a temporary directory](https://github.com/sstephenson/rbenv/blob/5b9e4f05846f6bd03b09b8572142f53fd7a46e62/test/test_helper.bash#L17) for the duration of the entire test suite. Any idea why this didn't happen for you? You could maybe add some debugging statements to the `--version` test case. Also, you can run a single test file instead of an entire suite— faster for debugging.

**Comment by jasonkarns** [comment]
I'm pretty sure it *did* run the suite in a temporary `$HOME`. But the setup is modifying global git config. From [test/--version.bats](https://github.com/sstephenson/rbenv/blob/5b9e4f05846f6bd03b09b8572142f53fd7a46e62/test/--version.bats#L7-L8):

```
setup() {
  mkdir -p "$HOME"
  git config --global user.name  "Tester"
  git config --global user.email "tester@test.local"
}
```

I assume the `--global` should just be removed? Like I said, I didn't investigate the fix much, other than removing those 2 lines from the setup function.


> Also, you can run a single test file instead of an entire suite— faster for debugging.

Yep. Though incidentally enough, I'm experiencing a completely unrelated [bug with Bats and single test-file execution](https://github.com/sstephenson/bats/issues/104). :)

**Comment by mislav** [comment]
> I assume the --global should just be removed?

`--global` means `$HOME/.gitconfig`.

**Comment by jasonkarns** [comment]
I have my system set up following the XDG basedir spec, so my git config is
under `$XDG_CONFIG_HOME/git/config`, not `$HOME`. `$XDG_CONFIG_HOME` is set
explicitly, but it's set to the default value `~/.config/` (though I
imagine the `$XDG_CONFIG_HOME` is saved after `~` is expanded, so changing
`$HOME` after setting `$XDG_CONFIG_HOME` would have no effect.

Speculating here. I can test this later.

**Comment by mislav** [comment]
Oh I see. Can you submit a pull request to reset XDG_CONFIG_HOME? Thanks

**Comment by mislav** [comment]
Fixes in 3a265c1
